src/uct/ib/mlx5/dc/dc_mlx5_devx.c

```diff
@@ -152,14 +153,16 @@ ucs_status_t uct_dc_mlx5_iface_devx_dci_connect(uct_dc_mlx5_iface_t *iface,
             uct_ib_mlx5_devx_set_qpc_port_affinity(md, dci_config->path_index,
                                                    qpc, &opt_param_mask);
         }
-
-        uct_ib_mlx5_devx_set_qpc_dp_ordering(
-                qpc, rc_iface->super.config.dp_ordering_ooo);
     } else {
         UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.sl,
                           rc_iface->super.config.sl);
     }
 
+    if (md->flags & UCT_IB_MLX5_MD_FLAG_DP_ORDERING_OOO_RW_DC) {
```

Comment:
seems the if() check is not needed as `...super.config.dp_ordering_ooo` is already including it.